docker/client.py

```diff
@@ -239,7 +241,7 @@ def _multiplexed_response_stream_helper(self, response):
         # Disable timeout on the underlying socket to prevent
         # Read timed out(s) for long running processes
         socket = self._get_raw_response_socket(response)
-        if six.PY3:
+        if six.PY3 and hasattr(socket, "_sock"):
```

Comment:
I did this because at this point all that matters is whether it has _sock or not, rather than what causes us to choose _sock inside _get_raw_response_socket.

It's possible though that we don't need the six.PY3 check in this case. I'll have a look at the behavior in Python2 later on today.